plugins/fileservices/fileservices.cpp

```diff
@@ -1233,28 +1266,39 @@ FILESERVICES_API char * FILESERVICES_CALL implementSprayJson(ICodeContext *ctx,
             req->setSrcPassword(userPw);
     }
 
-    Owned<IClientSprayResponse> result = server.SprayVariable(req);
-
-    StringBuffer wuid(result->getWuid());
-    if (!wuid.length())
+    OwnedSpanScope clientSpan = queryThreadedActiveSpan()->createClientSpan("Dfu Spray Json");
+    clientSpan->setSpanAttribute("destinationFilename", logicalName);
+    try
     {
-        const IMultiException* excep = &result->getExceptions();
-        if ((excep != NULL) && (excep->ordinality() > 0))
-        {
-            StringBuffer errmsg;
-            excep->errorMessage(errmsg);
-            throw MakeStringExceptionDirect(0, errmsg.str());
-        }
-        else
+        Owned<IClientSprayResponse> result = server.SprayVariable(req);
+
+        StringBuffer wuid(result->getWuid());
+        if (!wuid.length())
         {
-            throw MakeStringExceptionDirect(0, "Result's dfu WUID is empty");
+            const IMultiException* excep = &result->getExceptions();
+            if ((excep != NULL) && (excep->ordinality() > 0))
+            {
+                StringBuffer errmsg;
+                excep->errorMessage(errmsg);
+                throw MakeStringExceptionDirect(0, errmsg.str());
+            }
+            else
+            {
+                throw MakeStringExceptionDirect(0, "Result's dfu WUID is empty");
+            }
         }
-    }
 
-    wu.clear();
+        wu.clear();
 
-    blockUntilComplete("Spray", server, ctx, wuid, timeOut);
-    return wuid.detach();
+        clientSpan->setSpanAttribute("wuid", wuid);
+        blockUntilComplete("Spray", server, ctx, wuid, timeOut);
```

Comment:
I'm not familiar w/ this call, but by the name, it sounds like it could account for a significant amount of time, should that call be instrumented?
I have a call in hpcc4j which polls wsdfu for wu progress and periodically increases the wait time, is this a similar call?

Reply:
The time in this function will be included in the spans.  There were a couple I didn't adds spans for, but I will revisit in a separate PR.

Reply:
right, the new span will account for the total time, is it worth tracking the time spent  "blockingUntilComplete" ?

A separate PR seems appropriate, the question is, should the "blockUntilComplete" function declare its own span